### PR TITLE
INFOSSTRM-1466

### PR DESCRIPTION
**Pull request by jvnaiduwin**
Folderstructure match with uprotocol-sdk-java

### CODE REVIEW

**Comment by linsonittoop** [comment]
Please use a Pull Request heading relevant for the change: eg.. 'Folder restructuring'.

File: CMakeLists.txt [hunk 1]
```diff
@@ -155,4 +150,4 @@ target_link_libraries(uprotocolsdk_shared Threads::Threads
 add_dependencies(uprotocolsdk proto)
 add_dependencies(uprotocolsdk_shared proto)
 
-add_subdirectory(test)
\ No newline at end of file
+add_subdirectory(${CMAKE_SOURCE_DIR}/src/test)
```
Comment (linsonittoop) [inline CMakeLists.txt:153]: Avoid this change. keep the original line.

File: src/test/CMakeLists.txt [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
-        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-        DEPENDS "coverage"
-    )
-endif()
-
-
-include_directories(../proto)
+#if(CMAKE_COMPILER_IS_GNUCXX)
+#
+#	find_program(GCOV_TOOL gcov-11)
```
Comment (linsonittoop) [inline src/test/CMakeLists.txt:5]: instead of commenting this whole section, can we change it to gcov-9, if it would serve the purpose

**Review by linsonittoop — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,10 +1,3 @@
-set(VCPKG $ENV{HOME}/vcpkg/scripts/buildsystems/vcpkg.cmake)
```
[the remark this answers is not in the document]
Comment (jvnaiduwin) [reply CMakeLists.txt]: restored to existing code

File: CMakeLists.txt [hunk 4]
```diff
@@ -51,11 +44,11 @@ find_package(CGreen REQUIRED)
...
 
 set(PROTOS
-    ${CMAKE_CURRENT_SOURCE_DIR}/proto/cloudevents.proto	
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/uprotocol/proto/cloudevents.proto	
 )
-set(PROTO_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/proto)
-set(PROTO_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/proto)
-add_subdirectory(proto)
+set(PROTO_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/uprotocol/proto)
+set(PROTO_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/src/uprotocol/proto)
```
[the remark this answers is not in the document]
Comment (jvnaiduwin) [reply CMakeLists.txt]: Done

File: CMakeLists.txt [hunk 5]
```diff
@@ -51,11 +44,11 @@ find_package(CGreen REQUIRED)
...
+add_subdirectory(${CMAKE_SOURCE_DIR}/src/uprotocol/proto)
```
[the remark this answers is not in the document]
Comment (jvnaiduwin) [reply CMakeLists.txt]: Done

File: CMakeLists.txt [hunk 6]
```diff
@@ -86,19 +81,19 @@ if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
...
 endif()
 
-file(GLOB SRC_HDR_BINARY 	"include/binary/*.h")
-file(GLOB SRC_HDR_JSON 	 	"include/json/*.h")
-file(GLOB SRC_HDR_MODEL  	"include/model/*.h")
-file(GLOB SRC_HDR_FACTORY 	"include/factory/*.h")
-file(GLOB SRC_HDR_URI 		"include/uri/*.h")
-file(GLOB SRC_HDR_UUID 		"include/uuid/*.h")
-file(GLOB SRC_HDR_TOOLS 	"include/tools/*.h")
+file(GLOB SRC_HDR_BINARY 	"../uprotocol/serialize/*.h")
```
[the remark this answers is not in the document]
Comment (jvnaiduwin) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 1 above)
Comment (jvnaiduwin) [reply CMakeLists.txt]: Done

File: src/test/CMakeLists.txt (hunk 2 above)
Comment (jvnaiduwin) [reply src/test/CMakeLists.txt]: gcov-11 is working for me locally, restored to existing code

**Review by jvnaiduwin — COMMENTED** [review]
suggestion and comment resolved

**Review by linsonittoop — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (MelamudMichael) [reply CMakeLists.txt]: Hi @linsonittoop  @jvnaiduwin 
we dont need vcpkg

[the remark this answers is not in the document]
Comment (linsonittoop) [reply CMakeLists.txt]: When working on WSL, vcpkg is needed

[the remark this answers is not in the document]
Comment (jvnaiduwin) [reply CMakeLists.txt]: added Windows check

**Comment by jvnaiduwin** [comment]
new branch open

**Pull request closed without merging**